woocommerce/woocommerce-android
# Issue/5193 fetch single product review

Partially fixes #5193 

Merge Instructions:
1. Review [the PR in FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2192)
2. Review this PR
3. Merge FluxC
4. Update FluxC hash
5. Remove "Do not merge" label
5. Merge this PR

This PR refactors fetchSingleProductReview into a suspendable function - removes usage of the event bus.

There are two reasons for this change

1. [The event bus architecture is considered as legacy/deprecated](https://github.com/wordpress-mobile/WordPress-FluxC-Android/wiki/%5BDeprecated%5D-Architecture)
2. WCAndroid app uses stateful repositories which need to be registered to the event bus - this leads to memory leaks when the repository doesn't unregister from the event bus. It also causes side-effects when multiple instances of a repository are created - eg. duplicate tracking events.

To Test:
1. Put the app into background
2. Publish a new review on the web
3. Click on the "new review" notification on the mobile device
4. Verify the review detail loads
